tests/test.VmWatcher.js

```diff
@@ -0,0 +1,243 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2015, Joyent, Inc.
+ */
+
+var execFile = require('child_process').execFile;
+
+var test = require('tape');
+var vmadm = require('vmadm');
+
+var common = require('./common');
+var mocks = require('./mocks');
+var VmWatcher = require('../lib/vm-watcher');
+
+
+// For tests we can lower the frequency the periodic watcher polls so we finish
+// in more reasonable time.
+var PERIODIC_INTERVAL = 1000;
+// How frequently to poll the 'events' array when we're waiting for an event.
+var EVENTS_POLL_FREQ = 100; // ms
+
+var events = [];
+var existingVms = [];
+var smartosImageUUID;
+var smartosVmUUID;
+var watcher;
+
+
+function waitEvent(t, evt, vm_uuid, eventIdx) {
```

Comment:
Same than above, `waitEvent` should be factored out and reused by the different tests that current duplicate its implementation.